server/edp/server/test.py

```diff
@@ -0,0 +1,95 @@
+import cherrypy
+
+from girder.api import access
+from girder.api.describe import Description, autoDescribeRoute
+from girder.api.rest import Resource, RestException, boundHandler
+from girder.constants import AccessType, TokenScope
+from girder.models.file import File
+
+from girder.plugins.edp.models.experiment import Experiment as ExperimentModel
+from girder.plugins.edp.models.test import Test as TestModel
+
+
+@boundHandler
+@access.user(scope=TokenScope.DATA_WRITE)
+@autoDescribeRoute(
+    Description('Create a test.')
+    .modelParam('experimentId', 'The experiment id associated with this test',
+           model=ExperimentModel, destName='experiment',
+           level=AccessType.WRITE, paramType='path')
+    .jsonParam('test', 'The test', required=True, paramType='body')
+)
+def create(self, experiment, test):
+    self.requireParams(['cellId', 'channel'], test)
+
+    start_date = test.get('startDate')
+    cell_id = test.get('cellId')
+    channel = test.get('channel')
+    comments = test.get('comments')
+    public = test.get('public', False)
+
+    test = TestModel().create(experiment, start_date, cell_id,
+        channel, comments, self.getCurrentUser(), public)
+
+    cherrypy.response.status = 201
+    cherrypy.response.headers['Location'] = '/experiments/%s/tests/%s' % (experiment['_id'], test['_id'])
+
+    return test
+
+@boundHandler
+@access.user(scope=TokenScope.DATA_READ)
+@autoDescribeRoute(
+    Description('Get an test.')
+    .modelParam('experimentId', 'The experiment id associated with this test',
+        model=ExperimentModel, destName='experiment',
+        level=AccessType.READ, paramType='path')
+    .modelParam('testId', 'The test id',
+        model=TestModel, destName='test',
+        level=AccessType.READ, paramType='path')
+)
+def get(self, experiment, test):
+    return test
```

Comment:
Should we check for `test.experimentId == experiment['_id']` ?

Reply:
Yes, that is probably a good idea